core/src/main/java/com/datastax/oss/driver/internal/core/metadata/token/ReplicationFactor.java

```diff
@@ -0,0 +1,84 @@
+package com.datastax.oss.driver.internal.core.metadata.token;
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import com.datastax.oss.driver.shaded.guava.common.base.Preconditions;
+import com.datastax.oss.driver.shaded.guava.common.base.Splitter;
+import java.util.List;
+import java.util.Objects;
+
+// This class is a subset of server version at org.apache.cassandra.locator.ReplicationFactor
+public class ReplicationFactor {
+  private final int allReplicas;
+  private final int fullReplicas;
+
+  private ReplicationFactor(int allReplicas, int transientReplicas) {
+    this.allReplicas = allReplicas;
+    this.fullReplicas = allReplicas - transientReplicas;
+  }
+
+  public static ReplicationFactor withTransient(int allReplicas, int transientReplicas) {
```

Comment:
Not convinced by the utility of these factory methods vs usage of the raw constructor (+ maybe an overloading constructor that assigns 0 transient replicas).